app/controllers/reset_passwords_controller.rb

```diff
@@ -0,0 +1,18 @@
+class ResetPasswordsController < ApplicationController
+  def edit
+    @user = User.find_by token: params[:token]
+    @user ? render(:edit) : redirect_to(invalid_token_path)
+  end
+
+  def update
+    user = User.find_by token: params[:token]
+    if user
+      user.update_attribute :password, params[:user][:password]
```

Comment:
This pattern can be a problem as you dont know what user submitted as password. I mean if its invalid then you will provide him `'Successfully updated your password.'`.